api/models/log_events.py

```diff
@@ -33,46 +18,111 @@ class LogEvent(models.Model):
 
     date_time = models.DateTimeField(auto_now_add=True)
 
+    @property
+    def date_as_str(self) -> str:
+        return self.date_time.strftime("%d.%m.%Y")
+
     class Meta:
         abstract = True
 
 
+class GroupLogEvent(LogEvent):
+    class EventType(models.TextChoices):
+        FORMED = "formed", "Formed"
+        CONFIRMED = "confirmed", "Confirmed"
+        STARTED = "started", "Started classes"
+        FINISHED = "finished", "Finished classes"
+        # TODO put types here once they are finalized
+
+    group = models.ForeignKey(Group, on_delete=models.CASCADE)
+    type = models.CharField(
+        max_length=DEFAULT_CHOICE_CHAR_FIELD_MAX_LENGTH, choices=EventType.choices
+    )
+
+    def __str__(self):
+        return f"{self.date_as_str}: group {self.group} {self.get_type_display()}"
```

Comment:
`self.group()` will trigger additional DB hits because it will list teachers, coordinators and students... Maybe, just `self.group_id` will be enough

Reply:
I think we could think about that optimisation a little bit later. Maybe we will be able to query it with select_related to prevent additional db hits

Reply:
OK, "The root of all evil", huh? :)